### PR TITLE
Update Anvil and contribute multibindings for workers

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/414730916066338/1200105747936521/f
Tech Design URL: 
CC: 

**Description**:
This PR updates anvil to version 2.2.0. Anvil now provides a new handy annotation `ContributesMultibinding` that allow us to contribute dependencies into sets, like we do for eg. workers or view model factories. 
Using that annotation allow us to remove a bunch of boilerplate code.

In this PR we:
- Bumped anvil to 2.2.0 and upgrade dagger to 2.32
- Get VM factories to contribute multibinding
- Get Worker factories to contribute multibindings

**Steps to test this PR**:
1. app builds
2. install/upgrade app
3. verify bookmarks work
4. verify fireproof sites work
5. verify search/navigation works
6. verify fire button works

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: statistics/build.gradle [hunk 1]
```diff
@@ -41,10 +42,6 @@ dependencies {
 
     // Room
     implementation AndroidX.room.runtime
-    implementation AndroidX.room.rxJava2
-    kapt AndroidX.room.compiler
-    testImplementation AndroidX.room.testing
-    androidTestImplementation AndroidX.room.testing
```
Comment (aitorvs) [inline statistics/build.gradle:47]: We have room entities here but we don't need to have the room annotation processor here. We just need access to the annotations. The app module will take care of the code generation.

This should also improve build times a bit

File: versions.properties [hunk 2]
```diff
@@ -26,7 +26,7 @@ version.com.jakewharton.rxrelay2..rxrelay=2.0.0
 version.com.jakewharton.timber..timber=4.7.1
 version.com.nhaarman.mockitokotlin2..mockito-kotlin=2.1.0
 version.google.android.material=1.2.1
-version.google.dagger=2.27
+version.google.dagger=2.32
```
Comment (aitorvs) [inline versions.properties:29]: Updated dagger to have the same version than anvil 2.2.0.

File: statistics/build.gradle [hunk 3]
```diff
@@ -17,13 +17,14 @@
 plugins {
     id 'com.android.library'
     id 'kotlin-android'
-    id 'kotlin-kapt'
+    id 'com.squareup.anvil'
```
Comment (aitorvs) [inline statistics/build.gradle:20]: kapt is not needed here. it should improve a bit build times

**Review by malmstein — APPROVED** [review]
Updates and fresh installs working as expected, 💯 for removing code. Great stuff @aitorvs

**Pull request merged**